### PR TITLE
AMP-53112 Add code for initial storybook addon

### PR DESCRIPTION
**Pull request by maecapozzi**
<details>
  <summary>📦 Published PR as canary version: <code>0.0.3-canary.4.09c5d8f.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @amplitude/storybook-addon-amplitude@0.0.3-canary.4.09c5d8f.0
  # or 
  yarn add @amplitude/storybook-addon-amplitude@0.0.3-canary.4.09c5d8f.0
  ```
</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,16 @@
 {
-  "name": "storybook-addon-kit",
-  "version": "0.0.0",
-  "description": "everything you need to build a Storybook addon",
+  "name": "storybook-addon-amplitude",
+  "version": "0.0.1",
+  "description": "A storybook addon to capture events in Amplitude",
   "keywords": [
+    "storybook-addons",
```
Comment (jimmynotjim) [inline package.json:6]: nit, dupe keyword

File: package.json [hunk 2]
```diff
@@ -80,10 +80,14 @@
     "access": "public"
   },
   "storybook": {
-    "displayName": "Addon Kit",
+    "displayName": "storybook-addon-amplitude",
```
Comment (jimmynotjim) [inline package.json:83]: Should this be sentence case?

File: src/preset/manager.ts [hunk 3]
```diff
@@ -1,36 +1,44 @@
...
-addons.register(ADDON_ID, () => {
-  // Register the tool
-  addons.add(TOOL_ID, {
-    type: types.TOOL,
-    title: "My addon",
-    match: ({ viewMode }) => !!(viewMode && viewMode.match(/^(story|docs)$/)),
-    render: Tool,
-  });
+addons.register("storybook/amplitude", (api) => {
+  if (process.env.NODE_ENV === "production") {
```
Comment (jimmynotjim) [inline src/preset/manager.ts:9]: I don't know if this is even possible, but what do you think about this also taking an argument from the CLI as well to manually override? IDK that it'll matter much with storybook, but at my last job we'd often run into prod issues that were difficult to reproduce in local/staging envs due to hard coded prod-like conditionals.

**Review by jimmynotjim — COMMENTED** [review]
This is really cool. Just some comments for you to consider, but don't let it hold anything back.

File: src/preset/manager.ts (hunk 3 above)
Comment (maecapozzi) [reply src/preset/manager.ts:9]: Hm, I'll give this some thought. Not totally sure how I'd resolve this.

Comment (jimmynotjim) [reply src/preset/manager.ts:9]: Yeah, don't let it block anything, just something I've started paying attention to.

**Pull request merged**